clearblade/cloud/iot_v1/device_types.py

```diff
@@ -0,0 +1,429 @@
+        self._binary_data = binary_data
+
+    @property
+    def version_to_update(self):
+        return self._version_to_update
+
+    @property
+    def binary_data(self):
+        return self._binary_data
+
+class DeviceConfig(Request):
+    def __init__(self, name,
+                 version,
+                 cloud_ack_time,
+                 device_ack_time,
+                 binary_data) -> None:
+        super().__init__(name)
+        self._version = version
+        self._cloud_ack_time = cloud_ack_time
+        self._device_ack_time = device_ack_time
+        self._binary_data = binary_data
+
+    @property
+    def version(self):
+        return self._version
+
+    @property
+    def cloud_ack_time(self):
+        return self._cloud_ack_time
+
+    @property
+    def device_ack_time(self):
+        return self._device_ack_time
+
+    @property
+    def binary_data(self):
+        return self._binary_data
+
+    @staticmethod
+    def from_json(json):
+        return DeviceConfig(name='',
+                            version=get_value(json, 'version'),
+                            cloud_ack_time=get_value(json,'cloudUpdateTime'),
+                            device_ack_time=get_value(json, 'deviceAckTime'),
+                            binary_data=get_value(json,'binaryData'))
+
+class DeleteDeviceRequest(Request):
+    def __init__(self, name: str = None) -> None:
+        super().__init__(name)
+
+class GetDeviceRequest(Request):
+    def __init__(self, name: str = None) -> None:
+        super().__init__(name)
+
+class BindUnBindGatewayDeviceRequest(Request):
+    def __init__(self, parent:str = None,
+                 deviceId: str = None,
+                 gatewayId: str = None) -> None:
+        self._parent = parent
+        self._deviceid=deviceId
+        self._gatewayid=gatewayId
+
+    @property
+    def parent(self):
+        return self._parent
+
+    @property
+    def deviceId(self):
+        return self._deviceid
+
+    @property
+    def gatewayId(self):
+        return self._gatewayid
+
+class BindDeviceToGatewayRequest(BindUnBindGatewayDeviceRequest):
+    def __init__(self, parent: str = None,
+                 deviceId: str = None,
+                 gatewayId: str = None) -> None:
+        super().__init__(parent, deviceId, gatewayId)
+
+class UnbindDeviceFromGatewayRequest(BindUnBindGatewayDeviceRequest):
+    def __init__(self, parent: str = None,
+                 deviceId: str = None,
+                 gatewayId: str = None) -> None:
+        super().__init__(parent, deviceId, gatewayId)
+
+class ListDeviceStatesRequest(Request):
+    def __init__(self, name: str = None,
+                num_states: int = None) -> None:
+        super().__init__(name)
+        self._num_states = num_states
+
+    @property
+    def num_states(self):
+        return self._num_states
+
+class ListDeviceStatesResponse():
+    def __init__(self, device_states:list[DeviceState] = []) -> None:
```

Comment:
This line throws an error - `Type list not subscriptable`. Need to change list -> List and import List from typing.